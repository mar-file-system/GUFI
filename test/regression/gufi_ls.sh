#!/usr/bin/env bash

# This file is part of GUFI, which is part of MarFS, which is released
# under the BSD license.
#
#
# Copyright (c) 2017, Los Alamos National Security (LANS), LLC
# All rights reserved.
#
# Redistribution and use in source and binary forms, with or without modification,
# are permitted provided that the following conditions are met:
#
# 1. Redistributions of source code must retain the above copyright notice, this
# list of conditions and the following disclaimer.
#
# 2. Redistributions in binary form must reproduce the above copyright notice,
# this list of conditions and the following disclaimer in the documentation and/or
# other materials provided with the distribution.
#
# 3. Neither the name of the copyright holder nor the names of its contributors
# may be used to endorse or promote products derived from this software without
# specific prior written permission.
#
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
# ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
#
#
# From Los Alamos National Security, LLC:
# LA-CC-15-039
#
# Copyright (c) 2017, Los Alamos National Security, LLC All rights reserved.
# Copyright 2017. Los Alamos National Security, LLC. This software was produced
# under U.S. Government contract DE-AC52-06NA25396 for Los Alamos National
# Laboratory (LANL), which is operated by Los Alamos National Security, LLC for
# the U.S. Department of Energy. The U.S. Government has rights to use,
# reproduce, and distribute this software.  NEITHER THE GOVERNMENT NOR LOS
# ALAMOS NATIONAL SECURITY, LLC MAKES ANY WARRANTY, EXPRESS OR IMPLIED, OR
# ASSUMES ANY LIABILITY FOR THE USE OF THIS SOFTWARE.  If software is
# modified to produce derivative works, such modified software should be
# clearly marked, so as not to confuse it with the version available from
# LANL.
#
# THIS SOFTWARE IS PROVIDED BY LOS ALAMOS NATIONAL SECURITY, LLC AND CONTRIBUTORS
# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
# ARE DISCLAIMED. IN NO EVENT SHALL LOS ALAMOS NATIONAL SECURITY, LLC OR
# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
# OF SUCH DAMAGE.



set -e

ROOT="$(realpath ${BASH_SOURCE[0]})"
ROOT="$(dirname ${ROOT})"
ROOT="$(dirname ${ROOT})"
ROOT="$(dirname ${ROOT})"

# gufi_ls wrapper
GUFI_LS="${ROOT}/test/regression/gufi_ls.py"

# output directories
SRCDIR="prefix"
INDEXROOT="$(realpath ${SRCDIR}.gufi)"

source ${ROOT}/test/regression/setup.sh "${ROOT}" "${SRCDIR}" "${INDEXROOT}"

OUTPUT="gufi_ls.out"

function replace() {
    echo "$@" | sed "s/[[:space:]]*$//g; s/${GUFI_LS//\//\\/}/gufi_ls/g; s/.*${INDEXROOT//\//\\/}\\/$/./g; s/${INDEXROOT//\//\\/}\\///g"
}

function run() {
    replace "$ $@"
    replace "$($@)"
    echo
}

(
run "${GUFI_LS}"
run "${GUFI_LS} -a"
run "${GUFI_LS} -h"
run "${GUFI_LS} -R"
run "${GUFI_LS} -r"
run "${GUFI_LS} -ar"
run "${GUFI_LS} -arR"
) | tee "${OUTPUT}"

diff -b ${ROOT}/test/regression/gufi_ls.expected "${OUTPUT}"
rm "${OUTPUT}"
