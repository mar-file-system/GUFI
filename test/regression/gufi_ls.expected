$ gufi_ls
1KB
1MB
directory
empty_file
executable
file_symlink
leaf_directory
leaf_file1
leaf_file2
old_file
readonly
repeat_name
unusual, name?#
writable

$ gufi_ls -a
.hidden
1KB
1MB
directory
empty_file
executable
file_symlink
leaf_directory
leaf_file1
leaf_file2
old_file
readonly
repeat_name
unusual, name?#
writable

$ gufi_ls -h
1KB
1MB
directory
empty_file
executable
file_symlink
leaf_directory
leaf_file1
leaf_file2
old_file
readonly
repeat_name
unusual, name?#
writable

$ gufi_ls -R
1KB
1MB
directory
directory/executable
directory/readonly
directory/subdirectory
directory/subdirectory/directory_symlink
directory/subdirectory/repeat_name
directory/writable
empty_file
file_symlink
leaf_directory
leaf_directory/leaf_file1
leaf_directory/leaf_file2
old_file
repeat_name
unusual, name?#

$ gufi_ls -r
writable
unusual, name?#
repeat_name
readonly
old_file
leaf_file2
leaf_file1
leaf_directory
file_symlink
executable
empty_file
directory
1MB
1KB

$ gufi_ls -ar
writable
unusual, name?#
repeat_name
readonly
old_file
leaf_file2
leaf_file1
leaf_directory
file_symlink
executable
empty_file
directory
1MB
1KB
.hidden

$ gufi_ls -arR
unusual, name?#
repeat_name
old_file
leaf_directory/leaf_file2
leaf_directory/leaf_file1
leaf_directory
file_symlink
empty_file
directory/writable
directory/subdirectory/repeat_name
directory/subdirectory/directory_symlink
directory/subdirectory
directory/readonly
directory/executable
directory
1MB
1KB
.hidden

